### DISCUSSION TITLE
Optional Parameter in the Odata function

### DISCUSSION THREAD

**Discussion by imansafari1991** [bitfoundation/bitplatform#3308] [Q&A] [answered]
Hi 
I'm using bit v6.7
I want to use an optional parameter in the OData function
how can I do this?

Thanks

**Comment by yasmoradi** [comment, marked as answer]
This is not supported )":

**Reply by imansafari1991** [reply]
Thank you, Yaser
After a lot of tries, I found a solution, 
I am happy to share it while my eyes are bleeding 
I've just added the following few lines to GetODataModelBuilder Method:
`  EntitySetConfiguration<RoomSearchSeoDto> roomSearchSeo = builder.EntitySet<RoomSearchSeoDto>("RoomSearchSeo");
            FunctionConfiguration getroomSearchSeoOptional = roomSearchSeo.EntityType.Collection.Function("GetSearchInfo");
            getroomSearchSeoOptional.Parameter<string>("route");
            getroomSearchSeoOptional.Parameter<string>("code");
            getroomSearchSeoOptional.Parameter<string>("permalink").Optional();
            getroomSearchSeoOptional.Parameter<long>("houseType");
            getroomSearchSeoOptional.ReturnsFromEntitySet<RoomSearchSeoDto>("RoomSearchSeo");`
This worked for me :-)
